### ISSUE TITLE
Auto should have smoother movement when objects are close together in time

### ISSUE THREAD

**Issue by cl8n** [ppy/osu#9341]
"smoother movement" as in more linear instead of snapping to each note

streams look really wrong on Auto right now. I think this is an easy way to make Auto look a lot more natural

**Comment by peppy** [comment]
If you are interested in picking it up, this PR may interest you: https://github.com/ppy/osu/pull/3578

The code quality was too hard to follow therefore it hasn't been merged yet. It should be implemented as an alternative algorithm, choosable in ruleset settings, though.

**Comment by smoogipoo** [comment]
That PR relates to catch. Probably meant https://github.com/ppy/osu/pull/1962

**Comment by peppy** [comment]
that's the one. had trouble searching for it..

**Comment by Flutterish** [comment]
imho to make a natural auto we mostly need a "cursor momentum", so it takes time to slow down from certain movements and takes time to accelerate, like a human would. There is a very neat calculus solution ( I solved to make a smooth menu parralax for a game ) to calculate exactly at what time to reverse acceleration to end up at a given position with a given velocity as well.

**Comment by Flutterish** [comment]
see this: https://www.desmos.com/calculator/7bkiptlbf6
